dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/storage/impl/S3Utils.java

```diff
@@ -171,6 +183,25 @@ public String getResourceFileName(String tenantCode, String fileName) {
         return String.format(FORMAT_S_S, getS3ResDir(tenantCode), fileName);
     }
 
+    @Override
+    public String getResourceFileName(String fullName) {
+        // here is a quick fix here to get fileName. We get the resource upload path and
+        // get the index of the first appearance of resource upload path. The index is put
+        // in the start index of the substring function and get the result substring containing
+        // tenantcode and "resource" directory and the fileName.
+        // Then we split the result substring
+        // with "/" and join all elements except the first two elements because they are
+        // tenantCode and "resource" directory.
+        String resourceUploadPath =
+                RESOURCE_UPLOAD_PATH.endsWith("/") ? StringUtils.chop(RESOURCE_UPLOAD_PATH) : RESOURCE_UPLOAD_PATH;
+        // +1 because we want to skip the "/" after resource upload path as well.
+        String pathContainingTenantNResource = fullName.substring(
+                fullName.indexOf(resourceUploadPath)
+                        + resourceUploadPath.length() + 1);
+        String[] fileNameArr = pathContainingTenantNResource.split("/");
+        return Joiner.on("/").join(Arrays.stream(fileNameArr).skip(2).collect(Collectors.toList()));
+    }
```

Comment:
```suggestion
    public String getResourceFileName(String fullName) {
        // here is a quick fix here to get fileName. We get the resource upload path and
        // get the index of the first appearance of resource upload path. The index is put
        // in the start index of the substring function and get the result substring containing
        // tenantcode and "resource" directory and the fileName.
        // Then we split the result substring
        // with "/" and join all elements except the first two elements because they are
        // tenantCode and "resource" directory.
        String resourceUploadPath =
                RESOURCE_UPLOAD_PATH.endsWith(Constnats.FOLDER_SEPARATOR) ? StringUtils.chop(RESOURCE_UPLOAD_PATH) : RESOURCE_UPLOAD_PATH;
        // +1 because we want to skip the "/" after resource upload path as well.
        String pathContainingTenantNResource = fullName.substring(
                fullName.indexOf(resourceUploadPath)
                        + resourceUploadPath.length() + 1);
        String[] fileNameArr = pathContainingTenantNResource.split(Constnats.FOLDER_SEPARATOR);
        return Joiner.on(Constnats.FOLDER_SEPARATOR).join(Arrays.stream(fileNameArr).skip(2).collect(Collectors.toList()));
    }
```